pkg/services/sqlstore/dashboard.go

```diff
@@ -473,17 +473,28 @@ func deleteDashboard(cmd *models.DeleteDashboardCommand, sess *DBSession) error
 			}
 		}
 
-		// clean ngalert tables
-		ngalertDeletes := []string{
-			"DELETE FROM alert_rule WHERE namespace_uid = (SELECT uid FROM dashboard WHERE id = ?)",
-			"DELETE FROM alert_rule_version WHERE rule_namespace_uid = (SELECT uid FROM dashboard WHERE id = ?)",
-		}
+		if cmd.ForceDeleteFolderRules {
+			// clean ngalert tables
+			ngalertDeletes := []string{
+				"DELETE FROM alert_rule WHERE namespace_uid = (SELECT uid FROM dashboard WHERE id = ?)",
+				"DELETE FROM alert_rule_version WHERE rule_namespace_uid = (SELECT uid FROM dashboard WHERE id = ?)",
+			}
 
-		for _, sql := range ngalertDeletes {
-			_, err := sess.Exec(sql, dashboard.Id)
+			for _, sql := range ngalertDeletes {
+				_, err := sess.Exec(sql, dashboard.Id)
+				if err != nil {
+					return err
+				}
+			}
+		} else {
+			var existingRuleID int64
```

Comment:
nit (optional): You can do the existing check at the beginning and then simply check `if exists && ! cmd.ForceDeleteFolderRules` for a simpler logic that does not involve `if/else`